core/tx_pool.go

```diff
@@ -570,11 +570,7 @@ func (pool *TxPool) local() map[common.Address]types.Transactions {
 // rules and adheres to some heuristic limits of the local node (price and size).
 func (pool *TxPool) validateTx(tx *types.Transaction, local bool) error {
 	// Accept only legacy transactions until EIP-2718/2930 activates.
-	if !pool.eip2718 && tx.Type() != types.LegacyTxType {
-		return ErrTxTypeNotSupported
-	}
-	// Reject dynamic fee transactions until EIP-1559 activates.
-	if !pool.eip1559 && tx.Type() == types.DynamicFeeTxType {
+	if !pool.eip2718 {
```

Comment:
This patch just removes the `LegacyTxType`. @jdowning100 & @kiltsonfire have some independent work to remove unused EIP logic, which may or may not remove this `eip2718` check. I suspect it will go away, but that's for a different PR, IMO

Reply:
Yeah this if statement should be removed, but in a different PR